app/views/handlers/list_repeating_question.py

```diff
@@ -19,6 +19,18 @@ def get_previous_location_url(self):
         if url := self.get_section_or_final_summary_url():
             return url
 
+        edit_block = self._schema.get_edit_block_for_list_collector(
+            self.parent_block["id"]
+        )
+        if not edit_block:
+            block_id = self._request_args.get("previous")
+            return self._get_location_url(
+                block_id=block_id,
+                return_to=self._return_to,
+                return_to_answer_id=self._return_to_answer_id,
+                return_to_block_id=self._return_to_block_id,
+            )
```

Comment:
How come this is at this level?. This will skip the other logic routes if we don't have an edit block which is incorrect. This should be at the end. We also don't have a `previous` arg?. If we can't edit at the end, we should return to parent.

Reply:
Some confusion here, my mistake, moved it to the bottom now and using parent as previous, also removed type ignore and fixed some test that was incorrect.